priv/perf/apps/load_test/lib/common/sender_server.ex

```diff
@@ -0,0 +1,236 @@
+  def init({seqnum, utxo, ntx_to_send, opts}) do
+    defaults = [randomized: true]
+    opts = Keyword.merge(defaults, opts)
+
+    _ =
+      Logger.debug(
+        "[#{inspect(seqnum)}] init called with utxo: #{inspect(utxo)} and requests: '#{inspect(ntx_to_send)}'"
+      )
+
+    send(self(), :do)
+    {:ok, init_state(seqnum, utxo, ntx_to_send, opts)}
+  end
+
+  @doc """
+  Submits transaction then schedules call to itself if more left.
+  Otherwise unregisters from the Registry and stops.
+  """
+  @spec handle_info(:do, state :: __MODULE__.state()) ::
+          {:noreply, new_state :: __MODULE__.state()} | {:stop, :normal, __MODULE__.state()}
+  def handle_info(
+        :do,
+        %__MODULE__{ntx_to_send: 0, seqnum: seqnum, last_tx: %LastTx{blknum: blknum, txindex: txindex}} = state
+      ) do
+    _ = Logger.info("[#{inspect(seqnum)}] Stoping...")
+
+    LoadTest.Common.SenderManager.sender_stats(%{seqnum: seqnum, blknum: blknum, txindex: txindex})
+    {:stop, :normal, state}
+  end
+
+  def handle_info(:do, %__MODULE__{} = state) do
+    newstate =
+      state
+      |> prepare_and_submit_tx()
+      |> update_state_with_tx_submission(state)
+
+    {:noreply, newstate}
+  end
+
+  defp prepare_and_submit_tx(%__MODULE__{seqnum: seqnum, spender: spender, last_tx: last_tx, randomized: randomized}) do
+    to_spend = 1
+    new_amount = last_tx.amount - to_spend - @fees_amount
+
+    recipient =
+      if randomized do
+        {:ok, user} = Account.new()
+
+        user
+      else
+        spender
+      end
+
+    _ =
+      Logger.debug(
+        "[#{inspect(seqnum)}]: Sending Tx to new owner #{Base.encode64(recipient.addr)}, left: #{inspect(new_amount)}"
+      )
+
+    recipient_output = [%ExPlasma.Utxo{owner: recipient.addr, currency: @eth, amount: to_spend}]
+    # we aren't allowed to create zero-amount outputs, so if this is the last tx and no change is due, leave it out
+    change_output =
+      if new_amount > 0, do: [%ExPlasma.Utxo{owner: spender.addr, currency: @eth, amount: new_amount}], else: []
+
+    [%{blknum: blknum, txindex: txindex, amount: amount} | _] =
+      Transaction.submit_tx(
+        [%ExPlasma.Utxo{blknum: last_tx.blknum, txindex: last_tx.txindex, oindex: last_tx.oindex}],
+        change_output ++ recipient_output,
+        [
+          spender
+        ],
+        1_000
+      )
+
+    _ =
+      Logger.debug("[#{inspect(seqnum)}]: Transaction submitted successfully {#{inspect(blknum)}, #{inspect(txindex)}}")
+
+    {:ok, blknum, txindex, amount}
+  end
+
+  # Handles result of successful Tx submission or retry request into new state and sends :do message
+  @spec update_state_with_tx_submission(
+          tx_submit_result :: {:ok, map} | :retry | {:error, any},
+          state :: __MODULE__.state()
+        ) :: __MODULE__.state()
+  defp update_state_with_tx_submission(
+         tx_submit_result,
+         %__MODULE__{seqnum: seqnum, last_tx: last_tx} = state
+       ) do
```

Comment:
```suggestion
  defp update_state_with_tx_submission(tx_submit_result, state) do
```
and then use `.` access modifier (state.last_tx)